demos/demo_stacking_simulation.py

```diff
@@ -0,0 +1,1143 @@
+			x_init[:3] = np.resize(position,(3,1)) 
+			x_init[3:7] = np.resize(orientation,(4,1)) # quaternion
+			self.cube.reset_state(x_init)
+
+			initial_object_in_base_link = tf.compose_matrix(angles=(0, 0, 0),
+			                                        translate=(0, 0, 0.05))
+			self.object_tracker = ObjectTracker()
+			self.object_tracker.initialize(initial_object_in_base_link)
+
+		self.cube_q1 = self.get_cube_state()
+		self.cube_q_last = self.get_cube_state()
+
+
+	def get_desired_state(self):
+		# Calculating the desird position to reach
+		self.x_des = [np.matrix([self.object_position[0], self.object_position[1] + self.object_size/2, self.object_position[2] + 0.004]).T,
+				 np.matrix([self.object_position[0] + self.object_size/2, self.object_position[1] - self.object_size/8, self.object_position[2]]).T,
+				 np.matrix([self.object_position[0] - self.object_size/2, self.object_position[1] - self.object_size/8, self.object_position[2]]).T]
+
+		# self.x_des = [np.matrix([self.object_position[0], self.object_position[1] + self.object_size/2, self.object_position[2]]).T,
+		# 		 np.matrix([self.object_position[0] + self.object_size/2, self.object_position[1], self.object_position[2]]).T,
+		# 		 np.matrix([self.object_position[0] - self.object_size/2, self.object_position[1], self.object_position[2]]).T]
+
+		self.x_des = np.vstack(self.x_des)
+
+		self.v_des = [np.matrix([0,0,0]).T, np.matrix([0,0,0]).T, np.matrix([0,0,0]).T]
+		self.v_des = np.vstack(self.v_des)
+
+		# Kp and Kd
+		self.Kp = np.diag(np.full(9,81)) # np.diag(np.full(9,81))
+		self.Kd = np.diag(np.full(9,0.09)) # np.diag(np.full(9,0.09))
+
+
+
+	def get_torques(self):
+		'''
+		Torque = J.T * (F_desired + [kp*(x-x_des) + kd*(x`-x`_des)] )
+		'''
+		# Calculating total force at the end effector
+		# self.calculate_x_compensation()
+
+		force = self.Kp * (self.x_des - self.x_obs) + self.Kd * (self.v_des - self.v_obs)
+		# force = self.Kp * (self.x_des - self.fixed_values) + self.Kd * (self.v_des - self.v_obs)
+		
+		total_force = self.desired_external_force + force
+
+		torque = self.J.T * total_force
+		self.torque = np.array(torque).reshape(-1)
+
+
+	def calculate_q_compensation(self):
+		# obs = self.finger.frontend.get_observation(self.t)
+		obs = self.obs
+		q = obs.position.copy()
+		tau = obs.torque
+
+		if not self.is_grabbing:
+			return q
+
+
+		for i, (rad, torque) in enumerate(zip(q,tau)):
+			if torque > 0.03:
+				q[i] += math.radians(-1 * torque - 0.6)
+			elif torque < -0.1:
+				q[i] += math.radians(-1 * torque + 0.7)  
+			# else:
+			# 	q[i] += math.radians(-11 * (torque - 0.03) - 0.63)
+
+		return q
+
+
+
+	def get_next_state(self, debug=False):
+		'''
+		Returns the next state of the robot after applying the desired torque
+		'''
+		# Applying torque on real robot
+		if self.mode == "real":
+			time_stamp = self.finger.frontend.append_desired_action(self.finger.Action(torque=self.torque))
+			self.time_stamp = time_stamp
```

Comment:
Be careful with naming to avoid confusion.  This is not a time _stamp_ but a time _index_.